src/Repeater/DefaultRepeaterLauncher.ts

```diff
@@ -139,75 +100,59 @@ export class DefaultRepeaterLauncher implements RepeaterLauncher {
 
     logger.log('Starting the Repeater (%s)...', this.info.version);
 
-    this.repeaterId = repeaterId;
-
-    await this.bus.init();
+    this.repeater.connect();
 
-    const { payload }: RepeaterRegistered = await this.bus.send({
-      payload: new RepeaterRegistering(
-        repeaterId,
-        this.info.version,
-        !!this.virtualScripts.size
-      )
-    });
+    this.repeater.on(
+      'reconnection_failed',
+      this.onReconnectionFailed.bind(this)
+    );
+    this.repeater.on('request', this.onRequest.bind(this));
 
-    if ('error' in payload) {
-      this.statusChangeException(payload.error);
-    } else {
-      if (gt(payload.version, this.info.version)) {
-        logger.warn(
-          '%s: A new Repeater version (%s) is available, for update instruction visit https://docs.brightsec.com/docs/installation-options',
-          chalk.yellow('(!) IMPORTANT'),
-          payload.version
-        );
-      }
-
-      if (payload.script) {
-        this.compileScripts(payload.script);
-      }
-    }
+    const result = await this.repeater.deploy(repeaterId);
 
-    await this.subscribeToEvents();
+    this.repeaterId = result.repeaterId;
 
     this.repeaterStarted = true;
 
-    await this.bus.publish(
-      new RepeaterStatusUpdated(this.repeaterId, 'connected')
+    logger.log(
+      `The Repeater (%s) started. Repeater id is %s`,
+      this.info.version,
+      this.repeaterId
     );
+  }
 
-    logger.log(`The Repeater (%s) started`, this.info.version);
+  private onReconnectionFailed({
+    error
+  }: RepeaterReconnectionFailedEvent['request']) {
+    logger.error(error.message);
+    this.close();
+    process.exit(1);
   }
 
-  private async subscribeToEvents(): Promise<void> {
-    await this.bus.subscribe(RegisterScriptsHandler);
-    await this.bus.subscribe(NetworkTestHandler);
-    await this.bus.subscribe(SendRequestHandler);
-    this.timer = setInterval(
-      () =>
-        this.bus.publish(
-          new RepeaterStatusUpdated(this.repeaterId, 'connected')
-        ),
-      10000
```

Comment:
You have to send a ping each 10s otherwise repeater will be aborted by the backed

```ts
setInterval(() => socket.volatile.emit('ping'), 10000).unref();
```